hudi-utilities/src/main/java/org/apache/hudi/utilities/UtilHelpers.java

```diff
@@ -191,11 +192,14 @@ public static SchemaPostProcessor createSchemaPostProcessor(
 
   }
 
-  public static Option<Transformer> createTransformer(Option<List<String>> classNamesOpt, Boolean isErrorTableWriterEnabled) throws IOException {
+  public static Option<Transformer> createTransformer(Option<List<String>> classNamesOpt, Option<Schema> sourceSchema,
+                                                      boolean enableTransformerSchemaValidation, Boolean isErrorTableWriterEnabled) throws IOException {
+
     try {
-      return classNamesOpt.map(classNames -> classNames.isEmpty() ? null : 
-          isErrorTableWriterEnabled ? new ErrorTableAwareChainedTransformer(classNames) : new ChainedTransformer(classNames)
-      );
+      Function<List<String>, Transformer> chainedTransformerFunction = classNames ->
+          isErrorTableWriterEnabled ? new ErrorTableAwareChainedTransformer(classNames, sourceSchema, enableTransformerSchemaValidation)
```

Comment:
what is the expected behavior here if isErrorTableWriter is null?

Reply:
Addressed